docs/tutorials/customize_dataset.md

```diff
@@ -49,36 +49,90 @@ There are three necessary keys in the json file:
 - `categories`: contains the list of categories names and their ID.
 
-After the data pre-processing, the users need to further modify the config files to use the dataset.
-Here we show an example of using a custom dataset of 5 classes, assuming it is also in COCO format.
+After the data pre-processing, there are two steps for users to train the customized new dataset with existing format (e.g. COCO format):
+
+1. modify the config file for using the customized dataset.
+2. Check the annotations of the customized dataset.
+
+Here we give an example to show the above two steps, which uses a customized dataset of 5 classes with COCO format to train an existing Retinanet R50 FPN detector.
+
+#### 1. Modify the config file for using the customized dataset
+
+There are two aspects involved in the modification of config file:
+
+1. The `data` field. Specifically, you need to explicitly add the `classes` fields in `data.train`, `data.val` and `data.test`.
+2. The `num_classes` field in the `model` part. Explicitly over-write all the `num_classes` from default value (e.g. 80 in COCO) to your classes number.
 
 In `configs/my_custom_config.py`:
 
 ```python
-...
-# dataset settings
+
+# the new config inherits the base configs to highlight the necessary modification
+_base_ = './retinanet_r50_fpn_1x_coco.py'
+
+# 1. dataset settings
 dataset_type = 'CocoDataset'
 classes = ('a', 'b', 'c', 'd', 'e')
-...
 data = dict(
     samples_per_gpu=2,
     workers_per_gpu=2,
     train=dict(
         type=dataset_type,
+        # explicitly add your class names to the field `classes`
         classes=classes,
-        ann_file='path/to/your/train/data',
-        ...),
+        ann_file='path/to/your/train/annotation_data',
+        img_prefix='path/to/your/train/image_data'),
     val=dict(
         type=dataset_type,
+        # explicitly add your class names to the field `classes`
         classes=classes,
-        ann_file='path/to/your/val/data',
-        ...),
+        ann_file='path/to/your/val/annotation_data',
+        img_prefix='path/to/your/val/image_data'),
     test=dict(
         type=dataset_type,
+        # explicitly add your class names to the field `classes`
         classes=classes,
-        ann_file='path/to/your/test/data',
-        ...))
-...
+        ann_file='path/to/your/test/annotation_data',
+        img_prefix='path/to/your/test/image_data'))
+
+# 2. model settings
+# explicitly over-write all the `num_classes` field from default 80 to 5.
+model = dict(bbox_head=dict(num_classes=5))
+```
+
+#### 2. Check the annotations of the customized dataset
+
+Assuming your customized dataset is COCO format, make sure you have the correct annotations in the customized dataset:
+
+1. The length for `categories` field should exactly equal the number of classes (e.g. 5 in this example) in your config file.
+2. The `name` in `categories` field should have exactly the same order with `classes` field in your config file.
```

Comment:
Can we make sure this claim is correct?

Reply:
I tested my customized config based on a retinanet_r50_fpn model.
I guess at least the length for `categories` field should no less than the number of classes.

Reply:
At least the above are recommended and safe way to train the customized dataset with COCO format.

Reply:
Actually there is unnecessary for the `name` in `categories` field to have exactly the same order with `classes` field in the config file, as well as there should be exactly the same elements between them.
MMDetection automatically maps the uncontinuous `id` in `categories` to the continuous label indices. So it is the string order of `name` and `id` in `categories` field that matters, while the string order of `classes` in config does not matters.